src/pybind/mgr/orchestrator/module.py

```diff
@@ -424,24 +424,21 @@ def ukn(s):
             return HandleCommandResult(stdout=table.get_string())
 
     @_cli_write_command(
-        'orch osd create',
+        'orch apply osd',
```

Comment:
it seems like the apply command needs a name for the spec.  so you'd do something like
``ceph orch apply osd hdds -i hdds.drivegroup``, and it would show up as ``osd.hdds`` under ``ceph orch ls``?

Reply:
I was actually thinking to do it like that:

``` yaml
service_type: osd
drivegroups: 
  - hdd_dg:
      host_pattern: 'root2*'
      data_devices:
        rotational: 1
  - ssd_dg:
      host_pattern: 'room1*'
      data_devices:
        rotational: 0
```

which would still boil down to `ceph orch apply osd -i <list_of_drivegroups>`

This would internally be saved as `spec.osd` with value:

``` json

{
  "service_type": "osd", 
  "drivegroups": [
    {
      "hdd_dg": {
        "host_pattern": "root2*", 
        "data_devices": {
          "rotational": 1
        }
      }
    }, 
    {
      "ssd_dg": {
        "host_pattern": "room1*", 
        "data_devices": {
          "rotational": 0
        }
      }
    }
  ]
}

```

> and it would show up as osd.hdds under ceph orch ls?

The problem is that we currently don't have a good way to pin-point certain OSDs to a certain drivegroup. in `orch ls` we're only showing the total OSDs which makes it hard to say "this set of osds belongs to this drivegroup"..

However we could do:

``` yaml
service_type: osd
drive_group:
  name: ssds
  host_pattern: 'room1*'
  data_devices:
    rotational: 0
---

service_type: osd
drive_group:
  name: hdds
  host_pattern: 'room2*'
  data_devices:
    rotational: 1
```

which would be somewhat similar to `rgws` and would be saved as `osd.$drivegroup.name`

Reply:
i like that latter approach better since the dg name feels more natural.  I wonder, though, if the drive_group: nested is needed.. why not just
```
service_type: osd
name: ssds
host_pattern: 'room1*'
data_devices:
  rotational: 0
```

Reply:
the reason I nested it is because we may want to pass more information in there at some point. Also we may not only allow drivegroups but also plain devices specs?

Reply:
makes sense!